aws_auth.tf

```diff
@@ -59,6 +59,7 @@ locals {
         role["platform"] == "fargate" ? ["system:node-proxier"] : [],
       ))
     }
+    if role["worker_role_arn"] != ""
```

Comment:
I think we should instead return an empty list for `module.fargate.aws_auth_roles` https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/modules/fargate/outputs.tf#L21-L27.

Return [] if `local.create_eks`. Something like this by example:

```hcl
output "aws_auth_roles" {
  description = "Roles for use in aws-auth ConfigMap"
  value = [for i in range(1) : {
    worker_role_arn = local.pod_execution_role_arn
    platform        = "fargate"
    } if local.create_eks
  ]
}
```

Reply:
Yeah this might more elegant :) I've made this change with one exception and that's using `if length(var.fargate_profiles) > 0` instead of `local.create_eks` since it seems that `local.create_eks` would always be true and `var.fargate_profiles` seems to be control point for that submodule.

Reply:
Yes this seems more elegant actually, so I'll go with this! I'm just testing the change now.

Reply:
All working as I would expect!

Reply:
great.